File: solutions/10/10.js
function sumPrimesUntil(limit) {
    let is_prime = Array(limit).fill(1);

    let prime_sum = 0;
    for (let i = 2; i < limit; i++) {
        if (is_prime[i] == 1) {
            prime_sum += i;

            for (let j = i * 2; j < limit; j += i) {
                is_prime[j] = 0;
            }
        }
    }

    return prime_sum;
}

function main() {
    result = sumPrimesUntil(2000000);

    console.log(result);
}

main();

File: solutions/10/10.rs
fn sum_primes_until(limit: usize) -> usize {
    let mut is_prime = vec![1; limit];

    let mut prime_sum: usize = 0;
    for i in 2..limit {
        if is_prime[i] == 1 {
            prime_sum += i;

            for j in (i * 2..limit).step_by(i) {
                is_prime[j] = 0;
            }
        }
    }

    return prime_sum;
}

fn main() {
    let result: usize = sum_primes_until(2000000);

    println!("{}", result);
}
